src/infrasys/db_migrations.py

```diff
@@ -0,0 +1,221 @@
+        "metadata",
+    ]
+    cursor = conn.cursor()
+
+    cursor.execute(f"SELECT * FROM {TIME_SERIES_METADATA_TABLE} LIMIT 1")
+    columns = [desc[0] for desc in cursor.description]
+    if not all(column in columns for column in legacy_columns):
+        logger.error(f"Legacy schema does not match expected columns: {columns}")
+        msg = "Bug: Legacy schema doesn't match expected structure"
+        raise NotImplementedError(msg)
+
+    logger.debug("Creating backup tables.")
+    execute(
+        cursor,
+        f"ALTER TABLE {TIME_SERIES_METADATA_TABLE} RENAME TO {_LEGACY_METADATA_TABLE}",
+    )
+
+    logger.info("Creating new schema tables.")
+    execute(
+        cursor, f"CREATE TABLE {KEY_VALUE_STORE_TABLE}(key TEXT PRIMARY KEY, VALUE JSON NOT NULL)"
+    )
+    create_associations_table(connection=conn)
+
+    logger.info("Migrating data from legacy schema.")
+    cursor.execute(f"SELECT * FROM {_LEGACY_METADATA_TABLE}")
+    rows = cursor.fetchall()
+
+    sql_data_to_insert = []
+    for row in rows:
+        (
+            id_val,
+            time_series_uuid,
+            time_series_type,
+            initial_timestamp,
+            resolution,
+            name,
+            owner_uuid,
+            owner_type,
+            features_hash,
+            metadata_json,
+        ) = row
+
+        metadata = json.loads(metadata_json)
+
+        # Creating a flatten metadata from legacy schema.
+        serialized_info = {
+            "__metadata__": {
+                **{
+                    "quantity_metadata": metadata.get("quantity_metadata"),
+                    "normalization": metadata.get("normalization"),
+                },
+                **metadata[TYPE_METADATA].pop("fields", {}),
+            }
+        }
+
+        features_dict = {}
+        if metadata.get("user_attributes"):  # We renamed user_attributes to features
+            features_dict = metadata.pop("user_attributes")
+
+        owner_category = "Component"
+        length = metadata.get("length", 0)
+
+        # Old resolution was in timedelta format.
+        resolution = _str_timedelta_to_iso_8601(resolution)
+
+        # Fix for timestamp from: 2020-01-01 00:00 -> 2020-01-01T00:00
+        initial_timestamp = initial_timestamp.replace(" ", "T")
+        sql_data_to_insert.append(
+            {
+                "time_series_uuid": time_series_uuid,
+                "time_series_type": time_series_type,
+                "initial_timestamp": initial_timestamp,
+                "resolution": resolution,
+                "length": length,
+                "name": name,
+                "owner_uuid": owner_uuid,
+                "owner_type": owner_type,
+                "owner_category": owner_category,
+                "features_json": make_features_string(features_dict),
+                "serialization_info": json.dumps(serialized_info),
+                "uuid": str(uuid.uuid4()),  # metadata_uuid did not exist on tehe legacy
```

Comment:
if we leave it like uuid we do not have to manually parse it when deserializing.